ee/clickhouse/models/person.py

```diff
@@ -113,6 +113,36 @@ def delete_person(
     sync_execute(DELETE_PERSON_BY_ID, data)
 
 
+# counts duplicate distinct IDs in clickhouse for a team broken down by day
+def count_duplicate_distinct_ids_for_team(team_id: Union[str, int]) -> Dict:
+    query_result = sync_execute(
+        """
+        SELECT startdate, count(*)
+        FROM (
+            SELECT distinct_id, count(*) as count, toDate(min(timestamp)) as startdate
+            FROM (
+                SELECT person_id, distinct_id, max(_timestamp) as timestamp
+                FROM person_distinct_id
+                WHERE team_id = %(team_id)s
+                GROUP BY person_id, distinct_id, team_id
+                HAVING max(is_deleted) = 0
+            )
+            GROUP BY distinct_id
+            HAVING count > 1
+        )
+        GROUP BY startdate
+        """,
+        {"team_id": str(team_id)},
+    )
+    result = {}
+    total_duplicates = 0
+    for row in query_result:
+        total_duplicates += row[1]
+        result[row[0].isoformat()] = row[1]
```

Comment:
I wonder if it's worth sending data based on the distinct ID too (maybe a hash of it if you're concerned about PII), because there's no guarantee that users are repeated based on the date they were created. This is particularly useful because it's likely that users will have at least 2 distinct IDs (the anonymous and the identified), so it'll be hard for the total count to actually signal the problem correctly.

Reply:
So here we're thinking about one specific problem: ClickHouse having multiple IDs pointing to different persons, which this gives us insight into

Reply:
Having the distinct IDs (which are very sensitive PII imo) would only help to investigate the types of generic IDs people might use, but I'd rather do that analysis on Cloud

Reply:
How about a hash of the distinct ID, that way we can investigate better without compromising privacy

Reply:
Sorry, I'm a bit confused as to where you think having the distinct ID will help us.

Is it this?

> Having the distinct IDs (which are very sensitive PII imo) would only help to investigate the types of generic IDs people might use, but I'd rather do that analysis on Cloud

If so, while what I said it's true we'll mostly just get a lot of noise and it'll be complicated to do this analysis.

The duplicate IDs for team 1221 for example don't really tell us much. There's 100k+ of them but the problem is centered around 1: `distinctID`